contracts/MuonNodeStaking.sol

```diff
@@ -66,6 +66,22 @@ contract MuonNodeStaking is
     // tier => maxStakeAmount
     mapping(uint8 => uint256) public tiersMaxStakeAmount;
 
+    struct FunctionPauseState {
+        bool isPaused;
+    }
+    mapping(string => FunctionPauseState) public functionPauseStates;
+
+    /**
+     * @dev Modifier to make a function callable only when the contract is not paused.
+     */
+    modifier whenFunctionNotPaused(string memory functionName) {
+        require(
+            !functionPauseStates[functionName].isPaused,
```

Comment:
This is calling by all of the user functions. If we define constants and use uint8 as the mapping key, it wil be more gas efficient.

```
uint8 public constant FUNCTION_GET_REWARD = 1;
// ... other functions

... getReward() functionNotPaused(FUNCTION_GET_REWARD)
```